### PR TITLE
Add support for "AdditionalFiles" build action item type

### LINKED ISSUE #72: Project System: Add item build action defaults
_From @davkean on March 23, 2016 17:30_

Represents what happens when you add an item to a project, either by dragging it into the tree, including it, rename it or adding via Add Item dialog.

We need to set defaults, such as Build Action and Generator.

Make note that the WPF flavor has different and additional code to handle this above and beyond csproj.dll/msvbproj.dll.

_Copied from original issue: dotnet/roslyn#10011_

### PR DESCRIPTION
**Pull request by mavasani**
Fixes #72

### CODE REVIEW

**Comment by mavasani** [comment]
Tagging @dotnet/project-system for review.

**Comment by mavasani** [comment]
I verified that a project item's build action can be changed to "AdditionalFiles". What is the layer diagram scenario?

**Comment by srivatsn** [comment]
Just the screenshot in #72 - the project file is updated to be a `<AdditionalFiles Include="layer.diagram' />` and that file is passed to the compiler and LS as input

**Comment by mavasani** [comment]
I am unable to find an item template for layer diagram specifically, but I am able to add items to a web app and change their build action to AdditionalFiles, which adds the required item group to the project and passes it as an additional file to the command line compiler.

**Comment by srivatsn** [comment]
👍

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/ProjectItemsSchema.xaml [hunk 1]
```diff
@@ -58,9 +58,10 @@
       <NameValuePair Name="DefaultMetadata_Generator" Value="ResXFileCodeGenerator" />
     </ContentType>
 
+    <ItemType Name="AdditionalFiles" DisplayName="AdditionalFiles"/>
```
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/ProjectItemsSchema.xaml]: Can we give this a better display name? The others don't use compound pascal cased, they use sentence case. Also can we give a name that indicates it gets passed the compiler for an analyzer? Additional files is just too generate despite that's what we called the item.

**Comment by mavasani** [comment]
I think AdditionalFile is a term that is now widely used with analyzers. Even the item type, Roslyn public APIs and the compiler command line switch is called additionalfile. Any other term will actually cause more confusion.

**Comment by davkean** [comment]
It will be in the next public release of VS 2017.

**Comment by davkean** [comment]
@mavasani You work on the compiler team on analyzers, it's obvious to you - I've been on the team for nearly 2 years and only figured out what they were 6 months ago. I'm not asking to rename it, just provide more context in the display name.

On top of that, we use sentence case everywhere else in CPS for item type display names.

**Pull request merged**